bentoml/_internal/server/__init__.py

```diff
@@ -116,19 +118,19 @@ def serve_production(
     for runner_name, runner in svc.runners.items():
         sockets_path = os.path.join(uds_path, f"{id(runner)}.sock")
         assert len(sockets_path) < MAX_AF_UNIX_PATH_LENGTH
-
-        sockets_map[runner_name] = CircusSocket(
+        runner_socket = CircusSocket(
             name=runner_name,
             path=sockets_path,
             umask=0,
         )
+        sockets_map[runner_name] = runner_socket
         cmd_runner = f"""import bentoml._internal.server
 bentoml._internal.server.start_prod_runner_server(
     "{bento_identifier}",
     "{runner_name}",
     working_dir="{working_dir}",
     instance_id=$(CIRCUS.WID),
-    fd=$(circus.sockets.{runner_name}),
+    fd={runner_socket.fileno()},
```

Comment:
@bojiang I'm not sure why circus is not replacing it here, but changing it to `.fileno` call seems to be working well

Reply:
because we didn't ensure the runner_name to be a valid identifier.
Fixed in #2106